src/nupic/types/Sdr.hpp

```diff
@@ -498,42 +367,23 @@ class SparseDistributedRepresentation : public Serializable
      */
     template<typename T>
     void setSparse( const vector<vector<T>> &value ) {
-        NTA_ASSERT(value.size() == dimensions.size());
-        for(UInt dim = 0; dim < dimensions.size(); dim++) {
-            sparse[dim].clear();
-            for(auto itm: value[dim]) {
-                sparse[dim].push_back((UInt)itm);
-            }
-        }
-        setSparseInplace();
+      NTA_ASSERT(value.size() == dimensions.size());
+      for(UInt dim = 0; dim < dimensions.size(); dim++) {
+        sparse[dim].clear();
+        sparse[dim].assign(value[dim].cbegin(), value[dim].cend());
```

Comment:
I think this needs an explicit typecast, or else windows will squawk.

Reply:
Luckily no, our CI is happy with it.